clustering/operations.go

```diff
@@ -151,12 +151,36 @@ func (p *managerProcess) switchover(ctx context.Context, ss *StatusSet) error {
 	log.Info("begin switchover the primary", "current", ss.Primary, "next", ss.Candidate)
 
 	pdb := ss.DBOps[ss.Primary]
-	if err := pdb.SetReadOnly(ctx, true); err != nil {
-		return fmt.Errorf("failed to make instance %d read-only: %w", ss.Primary, err)
-	}
-	time.Sleep(100 * time.Millisecond)
-	if err := pdb.KillConnections(ctx); err != nil {
-		return fmt.Errorf("failed to kill connections in instance %d: %w", ss.Primary, err)
+
+	// SetReadOnly waits for a running DML.
+	// Therefore, if it waits for a long time, deleteGracePeriodSeconds may be reached.
+	// To avoid this, execute killConnections after a certain period of time.
+	done := make(chan error)
+	go func() {
+		done <- pdb.SetReadOnly(ctx, true)
+	}()
```

Comment:
This goroutine leaks if the control reaches line 169 because
the `done` channel is unbuffered.

I'd suggest making the channel buffered.
```suggestion
	done := make(chan error, 1)
	go func() {
		done <- pdb.SetReadOnly(ctx, true)
	}()
```